core/src/vaults.ts

```diff
@@ -0,0 +1,378 @@
+    const provider = await getProvider(network);
+    const liquidatorContract = new ethers.Contract(liquidatiorContractAddress, liquidatorContractInterface, provider);
+    const incentiveConstantDaiHex = await liquidatorContract.tip();
+    const incentiveRelativeDaiHex = await liquidatorContract.chip();
+    const incentiveRelativeDai = new BigNumber(incentiveRelativeDaiHex._hex)
+        .shiftedBy(-WAD_NUMBER_OF_DIGITS)
+        .multipliedBy(debtDai);
+    const incentiveConstantDai = new BigNumber(incentiveConstantDaiHex._hex).shiftedBy(-RAD_NUMBER_OF_DIGITS);
+    const incentiveCombinedDai = incentiveRelativeDai.plus(incentiveConstantDai);
+    return {
+        incentiveCombinedDai,
+        incentiveConstantDai,
+        incentiveRelativeDai,
+    };
+};
+
+export const fetchVaultLiquidationIncentive = memoizee(_fetchVaultLiquidationIncentive, {
+    maxAge: CACHE_EXPIRY_MS,
+    promise: true,
+    length: 3,
+});
+
+const _isVaultLiquidated = async (network: string, vault: Vault) => {
+    const contract = await getContract(network, 'MCD_DOG');
+    const typeHex = ethers.utils.formatBytes32String(vault.collateralType);
+    const eventFilter = contract.filters.Bark(typeHex, vault.address, null, null, null, null, null);
+    const liquidationEvents = await contract.queryFilter(eventFilter);
+    if (liquidationEvents.length !== 0 && vault.initialDebtDai.eq(0)) {
+        // there was a liquidation and the vault was not used again
+        const latestEvent = liquidationEvents[liquidationEvents.length - 1];
+        return {
+            isLiquidated: true,
+            liquidationDate: await fetchDateByBlockNumber(network, latestEvent.blockNumber),
+            transactionHash: latestEvent.transactionHash,
+            auctionId: new BigNumber(latestEvent.args?.id._hex).toFixed(),
+        };
+    }
+    return { isLiquidated: false };
+};
+export const isVaultLiquidated = memoizee(_isVaultLiquidated, {
+    maxAge: CACHE_EXPIRY_MS,
+    promise: true,
+    length: 2,
+});
+
+const _enrichVaultWithTransactonInformation = async (
+    network: string,
+    vault: Vault
+): Promise<VaultTransactionNotLiquidated> => {
+    const debtDai = vault.initialDebtDai.multipliedBy(vault.stabilityFeeRate);
+    const { liquidatiorContractAddress } = await fetchCollateralLiquidationLimitsAndLiquidatorAddress(
+        network,
+        vault.collateralType
+    );
+    const { incentiveCombinedDai, incentiveConstantDai, incentiveRelativeDai } = await fetchVaultLiquidationIncentive(
+        network,
+        liquidatiorContractAddress,
+        debtDai
+    );
+
+    const { liquidationRatio, oracleAddress } = await fetchLiquidationRatio(network, vault.collateralType);
+    const collateralizationRatio = vault.collateralAmount
+        .multipliedBy(vault.minUnitPrice)
+        .multipliedBy(liquidationRatio)
+        .dividedBy(debtDai)
+        .toNumber();
+    const proximityToLiquidation = collateralizationRatio - liquidationRatio;
```

Comment:
Would this stay within 0-1? I thought it will be not subtraction but division for some reason

Reply:
> Would this stay within 0-1?

nope, not guaranteed. 

> I thought it will be not subtraction but division for some reason

when i think about it more, i come to conclusion that it's confusing for the user to see both options (subtraction as it is done now and relative value). "Collateralization ratio" is itself a metric that is designed to compare against liquidation ratio: mathematically it means that regardless of what we do ( `-` or `/` ) it comes down to the comparison of the same values: `debtDai` vs `collateralAmount * minUnitPrice`. So i suggest instead just show this proximity as follows:

  - in dai
  - how much more debt should be there to make vault liquidatable (`collateralAmount * minUnitPrice - debtDai`)

Reply:
> in dai

This doesn't let the user _sort_ by the most risky vaults (something that we planned to do by default in the table). In your case, a small vault with 1.000 DAI "DAI proximity" will be much further from being liquidated than a vault with 10.000 DAI "DAI proximity". My suggestion is to keep it being a percentage: it can be the same value as you suggest by divided by the overall size of the vault in DAI: `(collateralAmount * minUnitPrice - debtDai) / collateralAmount * minUnitPrice`? (although I'm not sure if your formula is correct: don't we need to account for some fees/currentUnitPrice or debtDai is already derived from them?)

Reply:
made the change for this formula,

> although I'm not sure if your formula is correct: don't we need to account for some fees/currentUnitPrice or debtDai is already derived from them?

no, proximity to liquidation is not determined based on the fees, only on raw numbers of `initialDebt`, `stabilityFeeRate`, `collateralAmount`, `minUnitPrice`.  https://github.com/makerdao/dss/blob/fa4f6630afb0624d04a003e920b0d71a00331d98/src/dog.sol#L181